lib/matplotlib/collections.py

```diff
@@ -1254,6 +1255,264 @@ def set_verts_and_codes(self, verts, codes):
+            intersection point and extend the filled region up to this point.
+
+        step : {{'pre', 'post', 'mid'}}, optional
+            Define *step* if the filling should be a step function,
+            i.e. constant in between *t*.  The value determines where the
+            step will occur:
+
+            - 'pre': The f value is continued constantly to the left from
+              every *t* position, i.e. the interval ``(t[i-1], t[i]]`` has the
+              value ``f[i]``.
+            - 'post': The y value is continued constantly to the right from
+              every *x* position, i.e. the interval ``[t[i], t[i+1])`` has the
+              value ``f[i]``.
+            - 'mid': Steps occur half-way between the *t* positions.
+
+        **kwargs
+            Forwarded to `.PolyCollection`.
+
+        See Also
+        --------
+        .PolyCollection.set_verts, .Line2D.set_data
+        """
+        if not mpl.rcParams["_internal.classic_mode"]:
+            kwargs = cbook.normalize_kwargs(kwargs, Collection)
+
+        t, f1, f2 = self.axes._fill_between_process_units(
+            self.t_direction, self._f_direction, t, f1, f2, kwargs)
+
+        verts = self._make_verts(t, f1, f2, where, interpolate, step)
+        self.set_verts(verts)
+
+    def _make_verts(self, t, f1, f2, where, interpolate, step):
+        """
+        Make verts that can be forwarded to `.PolyCollection`.
+        """
+        self._validate_dimensions(self.t_direction, self._f_direction, t, f1, f2)
+
+        if where is None:
+            where = True
+        else:
+            where = np.asarray(where, dtype=bool)
+            if where.size != t.size:
+                raise ValueError(f"where size ({where.size}) does not match "
+                                 f"{self.t_direction} size ({t.size})")
+        where = where & ~functools.reduce(
+            np.logical_or, map(np.ma.getmaskarray, [t, f1, f2]))
+
+        t, f1, f2 = np.broadcast_arrays(np.atleast_1d(t), f1, f2, subok=True)
+
+        verts = [
+            self._make_verts_per_region(t, f1, f2, idx0, idx1, step, interpolate)
+            for idx0, idx1 in cbook.contiguous_regions(where)
+        ]
+
+        # Prepare the datalim for autoscale when called from _fill_between_x_or_y
+        if getattr(self, "axes", None) is None:
+            self._xys_for_datalim = self._normalize_pts(np.concatenate([
+                np.stack((t[where], f[where]), axis=-1) for f in (f1, f2)]))
+
+        return verts
+
+    @staticmethod
+    def _validate_dimensions(t_dir, f_dir, t, f1, f2):
+        """Validate dimensions of t- and f-data."""
+        names = (d + s for d, s in zip((t_dir, f_dir, f_dir), ("", "1", "2")))
+        for name, array in zip(names, [t, f1, f2]):
+            if array.ndim > 1:
+                raise ValueError(f"{name!r} is not 1-dimensional")
+
+    def _make_verts_per_region(self, t, f1, f2, idx0, idx1, step, interpolate):
```

Comment:
```suggestion
    def _make_verts_for_region(self, t, f1, f2, idx0, idx1, step, interpolate):
```
Slightly more precise, because this handles one region.

Reply:
428ef2fd52d7489cb4a1803242f6b9922a9afc74